server/core/src/main/scala/sttp/tapir/server/interpreter/ServerInterpreter.scala

```diff
@@ -220,9 +220,10 @@ class ServerInterpreter[R, F[_], B, S](
       val statusCode = outputValues.statusCode.getOrElse(defaultStatusCode)
 
       val headers = outputValues.headers
-      outputValues.body match {
-        case Some(bodyFromHeaders) => ServerResponse(statusCode, headers, Some(bodyFromHeaders(Headers(headers))), Some(output)).unit
-        case None                  => ServerResponse(statusCode, headers, None: Option[B], Some(output)).unit
+      (statusCode, outputValues.body) match {
+        case (StatusCode.NoContent | StatusCode.NotModified, Some(_)) => monad.error(new IllegalStateException(s"Unexpected response body when status code == $statusCode"))
```

Comment:
Maybe it would be better to say: `The status code XYZ doesn't allow a body`

Also ... what if the response body is defined as an `Option`? Or if the body is returned as an empty string `""` - shouldn't we special case on this?

Reply:
That would be hard to check in this general function. The body is of type `B` which is backend-dependent (like `akka.http.scaladsl.model.HttpEntity`).

Reply:
Ok, forget it then :)